code/modules/reagents/chemistry/machinery/chem_recipe_debug.dm

```diff
@@ -24,41 +24,69 @@
 	var/minorImpurity
 	///The count of reactions that resolve below 0.9 purity
 	var/majorImpurity
-	///If we failed to react this current chem so use a lower temp
+	///If we failed to react this current chem so use a lower temp - all reactions only
 	var/failed = 0
+	///If we're forcing optimal conditions
+	var/force_temp = FALSE
+	var/force_ph = FALSE
+	///Forced values
+	var/f_temp = 300
+	var/f_ph = 7
+	///Multiplier of product
+	var/vol_multi = 20
+	///If we're reacting
+	var/react = FALSE
+	///Number of delta times taken to react
+	var/react_time = 0
+	///IF we're doing EVERY reaction
+	var/process_all = FALSE
+	///The name
+	var/list/reaction_names = list()
+	///If it's started
+	var/reaction_stated = FALSE
+	///If we spawn a beaker at the end of a reaction or not
+	var/beaker_spawn = FALSE
+	///If we force min temp on reaction setup
+	var/min_temp = FALSE
 
 ///Create reagents datum
 /obj/machinery/chem_recipe_debug/Initialize()
 	. = ..()
 	create_reagents(9000)//I want to make sure everything fits
+	end_processing()
 
 ///Enable the machine
 /obj/machinery/chem_recipe_debug/attackby(obj/item/I, mob/user, params)
 	. = .()
-	if(processing)
+	ui_interact(usr)
+	/*if(processing)
 		say("currently processing reaction [index]: [cached_reactions[index]] of [cached_reactions.len]")
 		return
 	say("Starting processing")
 	setup_reactions()
-	begin_processing()
+	begin_processing()*/
 
 ///Enable the machine
 /obj/machinery/chem_recipe_debug/AltClick(mob/living/user)
 	. = ..()
 	if(processing)
 		say("currently processing reaction [index]: [cached_reactions[index]] of [cached_reactions.len]")
 		return
+	process_all = TRUE
 	say("Starting processing")
 	setup_reactions()
 	begin_processing()
 
 ///Resets the index, and creates the cached_reaction list from all possible reactions
 /obj/machinery/chem_recipe_debug/proc/setup_reactions()
 	cached_reactions = list()
-	for(var/V in GLOB.chemical_reactions_list)
-		if(is_type_in_list(GLOB.chemical_reactions_list[V], cached_reactions))
-			continue
-		cached_reactions += GLOB.chemical_reactions_list[V]
+	if(!process_all)
+		cached_reactions = reaction_names
+	else
+		for(var/V in GLOB.chemical_reactions_list)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Reply:
Sorry, this was a debug thing and badcode and my face is red.